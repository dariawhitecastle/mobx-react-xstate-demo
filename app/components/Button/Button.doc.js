import { describe, PropTypes } from 'react-desc'

import { genericProps, colorPropType } from '../../utils/propTypes'

export const doc = Button => {
  const DocumentedButton = describe(Button)
    .description('A button.')
    .details(
      `You can provide a single function child that will be called with
      'hover' and 'focus' keys. This allows you to customize the rendering
      of the Button in those cases.`,
    )
    .usage(
      `import { Button } from 'components/Button';
<Button primary label='Label' />`,
    )
    .intrinsicElement('button')

  DocumentedButton.propTypes = {
    ...genericProps,
    active: PropTypes.bool.description('Whether the button is active.').defaultValue(false),
    color: colorPropType.description(
      'Fill color for primary, label color for plain, border color otherwise.',
    ),
    disabled: PropTypes.bool.description('Whether the button is disabled.').defaultValue(false),
    fill: PropTypes.oneOfType([PropTypes.oneOf(['horizontal', 'vertical']), PropTypes.bool])
      .description('Whether the button expands to fill all of the available width and/or height.')
      .defaultValue(false),
    focusIndicator: PropTypes.bool
      .description("Whether when 'plain' it should receive a focus outline.")
      .defaultValue(true),
    hoverIndicator: PropTypes.oneOfType([
      PropTypes.bool,
      PropTypes.string,
      PropTypes.oneOf(['background']),
      PropTypes.shape({
        background: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
      }),
    ])
      .description(
        `The hover indicator to apply when the user is mousing over the
button. An object can be also be specified for color index support:
{background: 'neutral-2'}. This prop is meant to be used only
with plain Buttons.`,
      )
      .defaultValue(false),
    href: PropTypes.string.description('If specified, the button will behave like an anchor tag.'),
    target: PropTypes.oneOf(['_self', '_blank', '_parent', '_top']).description(
      `Specifies where to display the URL defined in the href property.`,
    ),
    icon: PropTypes.element.description('Icon element to place in the button.'),
    gap: PropTypes.oneOfType([
      PropTypes.oneOf(['xxsmall', 'xsmall', 'small', 'medium', 'large', 'xlarge']),
      PropTypes.string,
    ])
      .description(`The amount of spacing between icon and label in the button.`)
      .defaultValue('small'),
    label: PropTypes.node.description('Label text to place in the button.'),
    onClick: PropTypes.func.description(
      `Click handler. Not setting this property and not specifying a href
causes the Button to be disabled.`,
    ),
    plain: PropTypes.bool
      .description(
        `Whether this is a plain button with no border or pad.
Non plain button will show both pad and border.
The plain button has no border and unless the icon prop exist it has no pad as well.`,
      )
      .defaultValue(false),
    primary: PropTypes.bool
      .description(
        'Whether this is a primary button. There should be at most one per page or screen.',
      )
      .defaultValue(false),
    reverse: PropTypes.bool
      .description(
        `Whether an icon and label should be reversed so that the icon is at the
end of the anchor.`,
      )
      .defaultValue(false),
    type: PropTypes.oneOf(['button', 'reset', 'submit'])
      .description('The type of button. Set the type to submit for the default button on forms.')
      .defaultValue('button'),
    as: PropTypes.oneOfType([PropTypes.string, PropTypes.func]).description(
      `The DOM tag or react component to use for the element.`,
    ),
  }

  return DocumentedButton
}

export const themeDoc = {
  'global.hover.color': {
    description: 'The background color when hovering.',
    type: 'string',
    defaultValue: "{ dark: 'white', light: 'black' }",
  },
  'global.edgeSize.small': {
    description: 'The padding around an icon-only button.',
    type: 'string',
    defaultValue: '12px',
  },
  'global.colors.control': {
    description: 'The color of the border.',
    type: 'string | { dark: string, light: string }',
    defaultValue: "{ dark: 'accent-1', light: 'brand', }",
  },
  'global.colors.brand': {
    description: 'The light version of the border.',
    type: 'string',
    defaultValue: '#7D4CDB',
  },
  'global.colors.text': {
    description: 'The color of the text label.',
    type: 'string | { dark: string, light: string }',
    defaultValue: "{ dark: '#f8f8f8', light: '#444444' }",
  },
  'text.medium.size': {
    description: 'The font size of the text label.',
    type: 'string',
    defaultValue: '18px',
  },
  'text.medium.height': {
    description: 'The line height of the text label.',
    type: 'string',
    defaultValue: '24px',
  },
  'button.border.color': {
    description: `The color of the border.`,
    type: 'string | { dark: string, light: string }',
  },
  'button.border.radius': {
    description: 'The corner radius.',
    type: 'string',
    defaultValue: '18px',
  },
  'button.border.width': {
    description: 'The border width.',
    type: 'string',
    defaultValue: '2px',
  },
  'button.color': {
    description: `The color of the text label.`,
    type: 'string | { dark: string, light: string }',
  },
  'button.primary.color': {
    description: `The color of the background for primary buttons.`,
    type: 'string | { dark: string, light: string }',
  },
  'button.disabled.opacity': {
    description: 'The opacity when the button is disabled.',
    type: 'number',
    defaultValue: 0.3,
  },
  'button.padding.horizontal': {
    description: 'The horizontal padding.',
    type: 'string',
    defaultValue: '22px',
  },
  'button.padding.vertical': {
    description: 'The vertical padding.',
    type: 'string',
    defaultValue: '4px',
  },
  'button.extend': {
    description: 'Any additional style for the Button.',
    type: 'string | (props) => {}',
  },
}
