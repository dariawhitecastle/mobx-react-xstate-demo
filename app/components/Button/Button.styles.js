import styled from 'styled-components'
import { Button } from 'grommet'

export const StyledButton = styled(Button)`
  // border-radius: 3px;
`
