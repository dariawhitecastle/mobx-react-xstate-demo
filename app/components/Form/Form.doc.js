import { describe, PropTypes } from 'react-desc'

export const doc = Form => {
  const DocumentedForm = describe(Form)
    .description('A form that manages state for its fields.')
    .usage(
      `import { Form } from 'grommet';
<Form />`,
    )
    .intrinsicElement('form')

  DocumentedForm.propTypes = {
    errors: PropTypes.shape({})
      .description(
        `An object representing any errors in the data. They keys should
        match the keys in the value object.`,
      )
      .defaultValue({}),
    messages: PropTypes.shape({
      invalid: PropTypes.string,
      required: PropTypes.string,
    })
      .description('Custom validation messages.')
      .defaultValue({ invalid: 'invalid', required: 'required' }),
    onChange: PropTypes.func.description(
      'Function that will be called when any fields are updated.',
    ),
    onSubmit: PropTypes.func.description(
      `Function that will be called when the form is submitted. The
      single argument is an event containing the latest value object
      via \`event.value\`.`,
    ),
    onReset: PropTypes.func.description(
      `Function that will be called when the form is reset. The
      single argument is the event provided by react.`,
    ),
    value: PropTypes.shape({})
      .description('An object representing all of the data in the form.')
      .defaultValue({}),
  }

  return DocumentedForm
}
