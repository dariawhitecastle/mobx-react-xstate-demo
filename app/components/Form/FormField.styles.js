import styled from 'styled-components'
import { FormField } from 'grommet'
import colors from '../../utils/colors'

export const StyledFormField = styled(FormField)`
  input {
    height: 60px;
    padding-left: 20px;
    font-size: 16px;
    font-weight: normal;
    color: ${colors.primaryText};
  }

  label + div {
    border: solid 1px;
    border-radius: 4px;
    border-color: ${({ error }) => (error ? colors.coral : colors.mediumGrey)};
    &:focus-within {
      transition: 0.3s ease-in-out;
      border-color: ${colors.accent};
    }
  }

  div {
    border-bottom: none;
  }
  label {
    margin-left: 0;
    div {
      input {
        display: none;
      }
      div {
        margin-top: 20px;
        border: solid 1px ${colors.primaryText};
        svg {
          stroke: ${colors.primaryText};
        }
      }
    }
    span {
      padding-top: 20px;
      font-size: 16px;
      color: ${colors.primaryText};
    }
  }
`
