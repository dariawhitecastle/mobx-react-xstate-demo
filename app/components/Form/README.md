<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## createForm
A form that manages state for its fields.

## Usage

```javascript
import { Form } from 'grommet';
<Form />
```

## Properties

**errors**

An object representing any errors in the data. They keys should
        match the keys in the value object. Defaults to `{}`.

```
{

}
```

**messages**

Custom validation messages. Defaults to `{
  "invalid": "invalid",
  "required": "required"
}`.

```
{
  invalid: string,
  required: string
}
```

**onChange**

Function that will be called when any fields are updated.

```
function
```

**onSubmit**

Function that will be called when the form is submitted. The
      single argument is an event containing the latest value object
      via `event.value`.

```
function
```

**onReset**

Function that will be called when the form is reset. The
      single argument is the event provided by react.

```
function
```

**value**

An object representing all of the data in the form. Defaults to `{}`.

```
{

}
```
  
## Intrinsic element

```
form
```