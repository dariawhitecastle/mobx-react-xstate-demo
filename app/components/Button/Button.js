/* eslint-disable react/prop-types */
/**
 *
 * Button.js
 *
 * A common button
 */

import React, { forwardRef } from 'react'
import { Button } from 'grommet'

import { doc } from './Button.doc'
import helpers from '../../utils/helpers'

const createButton = forwardRef((props, ref) => {
  const {
    a11yTitle,
    active,
    alignSelf,
    as,
    color,
    disabled,
    fill,
    focusIndicator,
    gap,
    gridArea,
    hoverIndicator,
    href,
    icon,
    margin,
    onClick,
    plain,
    primary,
    reverse,
    target,
    title,
    type,
    ...rest
  } = props

  return (
    <Button
      a11yTitle={a11yTitle}
      active={active}
      alignSelf={alignSelf}
      as={as}
      color={color}
      disabled={disabled}
      fill={fill}
      focusIndicator={focusIndicator}
      gap={gap}
      gridArea={gridArea}
      hoverIndicator={hoverIndicator}
      href={href}
      icon={icon}
      label={title}
      margin={margin}
      onClick={onClick}
      plain={plain}
      primary={primary}
      ref={ref}
      reverse={reverse}
      target={target}
      type={type}
      {...rest}
    />
  )
})

export default helpers.createWithDoc({
  docFunction: doc,
  component: createButton,
})
