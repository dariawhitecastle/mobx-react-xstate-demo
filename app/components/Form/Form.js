/* eslint-disable react/prop-types */
/**
 * Form
 */

import React from 'react'
import { Form } from 'grommet'
import { doc } from './Form.doc'
import helpers from '../../utils/helpers'

const createForm = ({
  errors,
  messages,
  onChange,
  onReset,
  onSubmit,
  value,
  children,
  ...rest
}) => (
  <Form
    errors={errors}
    messages={messages}
    onChange={onChange}
    onReset={onReset}
    onSubmit={onSubmit}
    value={value}
    {...rest}
  >
    {children}
  </Form>
)

export default helpers.createWithDoc({
  docFunction: doc,
  component: createForm,
})
