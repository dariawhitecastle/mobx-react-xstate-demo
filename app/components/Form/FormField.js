/* eslint-disable react/prop-types */
/**
 * Form
 */

import React from 'react'
import { StyledFormField } from './FormField.styles'
import { doc } from './Form.doc'
import helpers from '../../utils/helpers'

const createFormField = props => {
  const {
    component,
    error,
    help,
    htmlFor,
    label,
    name,
    pad,
    required,
    validate,
    children,
    value,
    ...rest
  } = props
  return (
    <StyledFormField
      component={component}
      error={error}
      help={help}
      htmlFor={htmlFor}
      label={label}
      name={name}
      pad={pad}
      required={required}
      validate={validate}
      {...value}
      {...rest}
    >
      {children}
    </StyledFormField>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createFormField,
})
