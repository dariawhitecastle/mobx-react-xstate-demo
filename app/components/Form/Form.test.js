import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render, fireEvent } from '@testing-library/react'
import 'jest-styled-components'

import { Grommet } from 'grommet'
import Form from './Form'
import FormField from './FormField'
import { Button } from '../Button'

describe('Form', () => {
  afterEach(cleanup)

  test('empty', () => {
    const component = renderer.create(
      <Grommet>
        <Form />
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with field', () => {
    const component = renderer.create(
      <Grommet>
        <Form>
          <FormField name="test" />
        </Form>
      </Grommet>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('update', () => {
    const validate = jest
      .fn()
      .mockReturnValueOnce('too short')
      .mockReturnValueOnce(undefined)
    const onSubmit = jest.fn()
    const { getByPlaceholderText, getByText, container } = render(
      <Grommet>
        <Form onSubmit={onSubmit}>
          <FormField name="test" required validate={validate} placeholder="test input" />
          <FormField name="test2" placeholder="test-2 input" />
          <Button type="submit" primary label="Submit" />
        </Form>
      </Grommet>,
    )
    expect(container.firstChild).toMatchSnapshot()
    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: 'v' },
    })
    fireEvent.click(getByText('Submit'))
    expect(validate).toBeCalledWith('v', { test: 'v' })
    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: 'value' },
    })
    fireEvent.change(getByPlaceholderText('test-2 input'), {
      target: { value: 'value-2' },
    })
    fireEvent.click(getByText('Submit'))
    expect(validate).toBeCalledWith('value', {
      test: 'value',
      test2: 'value-2',
    })
    expect(onSubmit).toBeCalledWith(
      expect.objectContaining({ value: { test: 'value', test2: 'value-2' } }),
    )
  })

  test('regexp validation', () => {
    const onSubmit = jest.fn()
    const { getByPlaceholderText, getByText, queryByText } = render(
      <Grommet>
        <Form onSubmit={onSubmit}>
          <FormField
            name="test"
            required
            validate={{ regexp: /^[a-z]/i }}
            placeholder="test input"
          />
          <Button type="submit" primary label="Submit" />
        </Form>
      </Grommet>,
    )

    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: '1' },
    })
    fireEvent.click(getByText('Submit'))
    expect(getByText('invalid')).toMatchSnapshot()

    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: 'a' },
    })
    fireEvent.click(getByText('Submit'))
    expect(queryByText('invalid')).toBeNull()
  })

  test('required validation', () => {
    const onSubmit = jest.fn()
    const { getByPlaceholderText, getByText, queryByText } = render(
      <Grommet>
        <Form onSubmit={onSubmit}>
          <FormField name="test" required placeholder="test input" />
          <Button type="submit" primary label="Submit" />
        </Form>
      </Grommet>,
    )

    fireEvent.click(getByText('Submit'))
    expect(queryByText('required')).toMatchSnapshot()
    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: '1' },
    })
    expect(queryByText('required')).toBeNull()
  })

  test('reset clears form', () => {
    const onReset = jest.fn()
    const { getByPlaceholderText, getByText, queryByText } = render(
      <Grommet>
        <Form onReset={onReset}>
          <FormField name="test" required placeholder="test input" />
          <Button type="reset" primary label="Reset" />
        </Form>
      </Grommet>,
    )
    fireEvent.change(getByPlaceholderText('test input'), {
      target: { value: 'Input has changed' },
    })
    fireEvent.click(getByText('Reset'))
    expect(queryByText('Input has changed')).toBeNull()
  })

  test('initial values', () => {
    const onSubmit = jest.fn()
    const { getByText, queryByText } = render(
      <Grommet>
        {/* this test continues running forever if the whole event passed to onSubmit */}
        <Form onSubmit={({ value }) => onSubmit({ value })}>
          <FormField name="test" placeholder="test input" value={{ value: 'Initial value' }} />
          <FormField name="test2" value={{ value: 'Initial value2' }} />
          <Button type="submit" primary label="Submit" />
        </Form>
      </Grommet>,
    )
    fireEvent.click(getByText('Submit'))
    expect(queryByText('required')).toBeNull()
    expect(onSubmit).toBeCalledWith(
      expect.objectContaining({
        value: { test: 'Initial value', test2: 'Initial value2' },
      }),
    )
  })
})
