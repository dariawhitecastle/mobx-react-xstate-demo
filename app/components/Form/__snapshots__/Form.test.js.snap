// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form empty 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  />
</div>
`;

exports[`Form regexp validation 1`] = `
.c0 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
  color: #FF4040;
}

<span
  class="c0"
>
  invalid
</span>
`;

exports[`Form required validation 1`] = `
.c0 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
  color: #FF4040;
}

<span
  class="c0"
>
  required
</span>
`;

exports[`Form update 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c6 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c6:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c5 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
  border: none;
}

.c5::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c5::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c5::-moz-placeholder {
  color: #AAAAAA;
}

.c5:-ms-input-placeholder {
  color: #AAAAAA;
}

.c5::-moz-focus-inner {
  border: none;
  outline: none;
}

.c4 {
  position: relative;
  width: 100%;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c1 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(170,170,170);
}

.c1 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 div {
  border-bottom: none;
}

.c1 label {
  margin-left: 0;
}

.c1 label div input {
  display: none;
}

.c1 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c1 label div div svg {
  stroke: rgb(68,68,68);
}

.c1 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

<div
  class="c0"
>
  <form>
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <input
            autocomplete="off"
            class="c5"
            name="test"
            placeholder="test input"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <input
            autocomplete="off"
            class="c5"
            name="test2"
            placeholder="test-2 input"
            value=""
          />
        </div>
      </div>
    </div>
    <button
      class="c6"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`Form with field 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c5 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
  border: none;
}

.c5::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c5::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c5::-moz-placeholder {
  color: #AAAAAA;
}

.c5:-ms-input-placeholder {
  color: #AAAAAA;
}

.c5::-moz-focus-inner {
  border: none;
  outline: none;
}

.c4 {
  position: relative;
  width: 100%;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c1 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(170,170,170);
}

.c1 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 div {
  border-bottom: none;
}

.c1 label {
  margin-left: 0;
}

.c1 label div input {
  display: none;
}

.c1 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c1 label div div svg {
  stroke: rgb(68,68,68);
}

.c1 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="c1 c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <input
            autoComplete="off"
            className="c5"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            value=""
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
